src/langchain_google_alloydb_pg/async_vectorstore.py

```diff
@@ -826,11 +829,9 @@ async def aapply_vector_index(
             return
 
         # Create `alloydb_scann` extension when a `ScaNN` index is applied
-        if isinstance(index, ScaNNIndex):
-            async with self.engine.connect() as conn:
-                await conn.execute(text("CREATE EXTENSION IF NOT EXISTS alloydb_scann"))
-                await conn.commit()
-            function = index.distance_strategy.scann_index_function
+        if index.enable_extension:
+            if callable(getattr(index, "create_extension", None)):
+                function = await index.create_extension(self.engine)  # type: ignore
```

Comment:
I want to keep all engine connection code in the async_vectorstore vs the index classes. I think we can use extension_name attribute as a flag instead of enable_extension. In the baseindex we would set extension_name = None and in the scann class extension_name= alloydb_scann
```suggestion
        if index.extension_name:
            async with self.engine.connect() as conn:
                await conn.execute(text("CREATE EXTENSION IF NOT EXISTS {index.extension_name}"))
                await conn.commit()
```

Reply:
Removed the usage of enable_extension and added extension_name